src/dns/server.rs

```diff
@@ -347,15 +347,15 @@ impl Store {
                         .unwrap();
 
                     return Some(ServerMatch {
-                        server: Address::Service(Box::new(service)),
+                        server: Address::Service(Arc::new(service)),
```

Comment:
In general for most of these heavy types we should probably just implement them with constructors that prewrap them, or over time we will probably end up with inconsistent usage.

Reply:
We actually are storing them as Arc and then clone them in the state.rs. So most of this is stoppnig to do that. I didn't tackle every codepath yet but presumably we could